src/Lifecycle.js

```diff
@@ -307,7 +309,7 @@ export function setLoggedIn(credentials) {
  * @param {MatrixClientCreds} credentials
  * @param {Boolean} clearStorage
  *
- * returns a Promise which resolves once the client has been started
+ * @returns {Promise} promise which resolves to the new MatrixClient once it has been started
```

Comment:
`startMatrixClient` calls `matrixClient.store.startup` and `startClient`, which are async. So it won't have finished starting... I'm not sure if the distinction is useful but someone might assume that the store and sync have been started once the promise resolves.